openedx/core/djangoapps/content/search/views.py

```diff
@@ -29,6 +36,38 @@ def _get_meili_api_key_uid():
     return _get_meili_api_key_uid.uid
 
 
+def _get_meili_access_filter(request: Request) -> dict:
+    """
+    Return meilisearch filter based on the requesting user's permissions.
+    """
+    # Global staff can see anything, so no filters required.
+    if GlobalStaff().has_user(request.user):
+        return {}
+
+    # Everyone else is limited to their org staff roles...
+    user_orgs = _get_user_orgs(request)[:MAX_ORGS_IN_FILTER]
+
+    # ...or the N most recent courses and libraries they can access.
+    access_ids = get_access_ids_for_request(request)[:MAX_ACCESS_IDS_IN_FILTER]
```

Comment:
Do you think it's worth passing in the list of orgs so that we can exclude any courses for which the user has an org-level role? It would make this slightly more complex but I think it could be a worthwhile optimization.

Reply:
Great idea, thank you! It wasn't hard to do: [811687d](https://github.com/openedx/edx-platform/pull/34471/commits/811687d9f4271793bb786d9e2e31eca769b7c599)